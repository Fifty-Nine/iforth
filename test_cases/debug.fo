: foo 1 2 3 .d ;
: bar foo drop .d ;
foo bar 0
clear
.d
