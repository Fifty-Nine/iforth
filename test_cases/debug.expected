========= machine state =========
token stream:
0:[:] 1:[foo] 2:[1] 3:[2] 4:[3] 5:[.d] 6:[;] 7:[:] 8:[bar] 9:[foo] 10:[drop] 11:[.d] 12:[;] 13:[foo] 14:[bar] 15:[0] 16:[clear] 17:[.d] 

data stack:
[2:1 1:2 0:3]

return stack:
[0:14]

ip: 5 (.d)
=================================
========= machine state =========
token stream:
0:[:] 1:[foo] 2:[1] 3:[2] 4:[3] 5:[.d] 6:[;] 7:[:] 8:[bar] 9:[foo] 10:[drop] 11:[.d] 12:[;] 13:[foo] 14:[bar] 15:[0] 16:[clear] 17:[.d] 

data stack:
[5:1 4:2 3:3 2:1 1:2 0:3]

return stack:
[1:15 0:10]

ip: 5 (.d)
=================================
========= machine state =========
token stream:
0:[:] 1:[foo] 2:[1] 3:[2] 4:[3] 5:[.d] 6:[;] 7:[:] 8:[bar] 9:[foo] 10:[drop] 11:[.d] 12:[;] 13:[foo] 14:[bar] 15:[0] 16:[clear] 17:[.d] 

data stack:
[4:1 3:2 2:3 1:1 0:2]

return stack:
[0:15]

ip: 11 (.d)
=================================
========= machine state =========
token stream:
0:[:] 1:[foo] 2:[1] 3:[2] 4:[3] 5:[.d] 6:[;] 7:[:] 8:[bar] 9:[foo] 10:[drop] 11:[.d] 12:[;] 13:[foo] 14:[bar] 15:[0] 16:[clear] 17:[.d] 

data stack:
[]

return stack:
[]

ip: 17 (.d)
=================================
